celery/canvas.py

```diff
@@ -1272,6 +1273,7 @@ class chord(Signature):
     def from_dict(cls, d, app=None):
         options = d.copy()
         args, options['kwargs'] = cls._unpack_args(**options['kwargs'])
+        args = [maybe_signature(task) for task in args]
```

Comment:
My more complex failure case hits this code somewhere but it doesn't seem to be required to fix my breakage (just the change to groups). In any case, this seems like a legit change to make.

Reply:
Could you please include the more complicated cases as tests in our integration suite?
You can mark them as `xfail` for now.

Reply:
Turns out this isn't strictly necessary because the `_maybe_group()` and `_maybe_signature()` in `chord.__init__()` cause nested `dicts` to get deserialized properly.

https://github.com/celery/celery/blob/9b5f6f5531f45525e904cef114d653a93f1c9635/celery/canvas.py#L1289-L1298

It's slightly different behaviour to the other `from_dict()` code paths since `chain.from_dict()` and (now) `group.from_dict()` deserialize their children prior to instantiating new objects. I've removed this from the diff, but let me know if you'd like it restored to make all three behave in the same way. (/ping @thedrow )